feat(back): 유저 로그인 일부 수정(#9)

bsy3764 opened the pull request:
feat: user service, controller, dto update

Commit:
feat: user service, controller, dto update

Commit:
Merge branch 'main' into back/feature/#9-유저-로그인

JuHyun419 commented on back/src/main/java/com/github/oneline/onelinecourse/controller/user/UserController.java:
```diff
@@ -18,20 +22,33 @@ public UserController(UserService userService) {    // DI 의존성 주입....
+                userService.searchUser(userId)
+                .stream()
+                .map(ResponseUserDto::new)
+                .collect(Collectors.toList()));
```
userId는 Unique한 값이라서 userId로 회원을 존재하면 최대 한명만 존재하지 않을까요??

JuHyun419 commented on back/src/main/java/com/github/oneline/onelinecourse/controller/user/UserController.java:
```diff
@@ -18,20 +22,33 @@ public UserController(UserService userService) {    // DI 의존성 주입....
         // 응답 헤더의 상태 코드 본문을 직접 다루기 위해 사용
         return ResponseEntity.ok(   // 200 ok 상태코드 설정
-                new ResponseUserDto(userService.save(createUserRequestDto.toEntity()))
+                new ResponseUserDto(userService.createUser(createUserRequestDto.toEntity()))
```
검증 조건이 필요할 것 같아요, 요청온 회원이 등록이 되어있는데 다시 등록을 하면 안되겠죠??

JuHyun419 commented on back/src/main/java/com/github/oneline/onelinecourse/repository/user/UserRepository.java:
```diff
@@ -2,12 +2,15 @@
 public interface UserRepository extends JpaRepository<User, String> {
-
+    public List<User> findByUserId(String userId);
+    public List<User> findByPlatform(String platform);
```
이 메서드는 혹시 어디서 사용이 되는걸까요??

Commit:
fix: search user modify(List<User> -> User)
and Merge branch 'main' of https://github.com/JuHyun419/one-line-course into back/feature/#9-유저-로그인

Commit:
fix: user model, repo, service
- controller의 serivce file 삭제

Commit:
feat: login user check
post요청의 유저가 최초 로그인 여부 확인
